### ISSUE TITLE
[QUESTION] What is needed from device to ensure tiles are visible

### ISSUE THREAD

**Issue by marcelrv** [cagnulein/qdomyos-zwift#2791]
Hi Roberto,
I wrote the device for the LifeFitness Elliptical based on CSAFE.

It seems to work fine now as birtual bluetooth device, values are visible and updating as expected.
Hoever, the tiles are not visible. I can't seem to find what is missing in the device (csaveelliptical) or other setting/file.
Hope you can give a hint

https://github.com/marcelrv/qdomyos-zwift/tree/dev

If I select e.g. fakeellipical as device, the tiles are rendering fine... 
I guess it is something in my driver, but is not obvious to me what is missing.

thx

[debug-zo_nov__17_14_54_17_2024.log](https://github.com/user-attachments/files/17790744/debug-zo_nov__17_14_54_17_2024.log)

**Comment by cagnulein** [comment]
I'm outside, didn't check the code: did you add the device in the
Bluetooth.cpp mega if then else? I suggest to add a csafe rower setting for
this

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 17 nov 2024 alle 15:01 Marcel ***@***.***> ha
scritto:

**Comment by marcelrv** [comment]
Yes, I made a copy of the csaferower in bluethooth.cpp as starting point, 
indeed it is next to the csaferower in the mega :-)  if/then/else

**Comment by cagnulein** [comment]
So are you entering on that entry? If not That's the reason why you don't
see the tiles

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 17 nov 2024 alle 15:23 Marcel ***@***.***> ha
scritto:

**Comment by marcelrv** [comment]
No, both entries are there. There is a separate config option `csafeellipticalSerialPort` that it uses
The loading of the device does not seem to be the issue, (the object  csafeElliptical = new csafeelliptical is created and running )

https://github.com/marcelrv/qdomyos-zwift/blob/dev/src/devices/bluetooth.cpp#L775-L807

**Comment by cagnulein** [comment]
I will  check your implementation in the next days

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno dom 17 nov 2024 alle 19:09 Marcel ***@***.***> ha
scritto:

**Comment by cagnulein** [comment]
sorry i should put this on the class but i did in this way very long time ago and i never find the time to fix the driver implementation. usually i just search for a device type and i'm putting the code on each point

**Comment by marcelrv** [comment]
I'm surprised I missed it actually. 
I did several times a search for csaferower to make sure I added an equivalent one for the csafeelliptical...  
seems that I everytime overlooked this one.
 
btw.. I saw few 2 issues in the csafe rower.. will make a PR for that as well. (don't think the current code actually works)
* the `deviceFilename` variable is not updated, hence a blank port is being opened
* metatype issue: because of `void onStatus(uint16_t status);` 
```Warning:   QObject::connect: Cannot queue arguments of type 'uint16_t'
(Make sure 'uint16_t' is registered using qRegisterMetaType().)
QObject::connect: Cannot queue arguments of type 'uint16_t'
(Make sure 'uint16_t' is registered using qRegisterMetaType().)```
